Jetpack App (Emphasis): Add a new JetpackScreenshotGeneration target

momo-ozawa opened the pull request:
Part of  #16395 

This PR adds a new JetpackScreenshotGeneration target.

## Description
- Duplicated the WordPressScreenshotGeneration target
- Added UI tests for App Store screenshot screens (Design ref: pcdRpT-mz-p2)
    - My Site
    - Activity Log
    - Scan
    - Backup
    - Stats 

## How to test

Note: My plan is to update WordPressMocks in a separate PR, hence disabling stubbing in the testing directions in this PR.

1. Disable WireMock by removing it from UI Test launch arguments `XCTest+Extensions.swift:73`
```swift
app.launchArguments = ["-wpcom-api-base-url", "-no-animations", "-ui-testing"]
// app.launchArguments = ["-wpcom-api-base-url", WireMock.URL().absoluteString, "-no-animations", "-ui-testing"]
```
2. Replace credentials on `JetpackScreenshotGeneration.swift:25` with credentials for an account that has a site where Backup and Scan are enabled (i.e. Jetpack Complete test site)
3. If needed, add the following lines in `JetpackScreenshotGeneration.swift` to make sure you're switching to a site that has both Backup and Scan
```swift
// Get My Site screenshot
let mySite = MySiteScreen()
    .showSiteSwitcher()
    .switchToSite(withTitle: <siteAddress>)
    .thenTakeScreenshot(1, named: "MySite")
```
4. ⚠️ Delete any existing Jetpack apps from your simulator
5. ⚠️ Make sure to disconnect the hardware keyboard in our simulator settings (Cmd+Shift+K)... otherwise the tests will fail
4. Finally, test the `JetpackScreenshotGeneration` scheme
5. ✅ `testGenerateScreenshots` should pass


## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Duplicate WordPressScreenshotGeneration target

Commit:
Set the JetpackScreenshotGeneration target app to Jetpack

Commit:
Update product bundle ID to com.jetpack.JetpackScreenshotGeneration

Commit:
Add a new JetpackScreenshotGeneration scheme

Commit:
Enable buildForRunning in the JetpackScreenshotGeneration scheme

Edit Scheme > Build > enable the checkbox under the Run column

Commit:
Remove WordPress from target dependencies

Commit:
Add new JetpackScreeshotGeneration UI test class

Commit:
Remove WordPressScreenshotGeneration test class from JetpackScreenshotGeneration target

Commit:
Add email and passsword login flow

Commit:
Add an activity log screen

Commit:
Show the activity log screen

Commit:
Add a Jetpack scan screen

Commit:
Show the Jetpack scan screen

Commit:
Add Jetpack back and backup options screens

Commit:
Show the Jetpack backup screen

Commit:
Merge branch 'develop' into task/16395-jetpack-screenshot-generation

# Conflicts:
#	WordPress/WordPress.xcodeproj/project.pbxproj

leandroalonso reviewed (changes requested):
@momo-ozawa works as described!

The only issue is that `WordPressScreenshotGeneration` is now failing to build.

Commit:
Add new UI test screens to WordPressScreenshotGeneration target

- ActivityLogScreen
- JetpackScanScreen
- JetpackBackupScreen
- JetpackBackupOptionsScreen

momo-ozawa commented:
@leandroalonso Thanks for reviewing! 

> The only issue is that WordPressScreenshotGeneration is now failing to build.

Fixed in 153fc7f 👍
